packages/interface/src/components/ui/Form.tsx

```diff
@@ -227,16 +228,25 @@ export const Form = <S extends z.Schema>({
     resolver: zodResolver(schema),
     mode: "onBlur",
   });
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
   // Pass the form methods to a FormProvider. This lets us access the form from components with useFormContext
   return (
     <FormProvider {...form}>
       <form
-        onSubmit={form.handleSubmit((data) => {
-          onSubmit(data, form);
-        })}
+        onSubmit={form.handleSubmit(
+          (data) => {
+            setErrorMessage(null);
+            onSubmit(data, form);
+          },
+          () => {
+            setErrorMessage("There are errors in the form. Please go back and check the warnings.");
+          },
```

Comment:
handlers should not be inside jsx render. Use callbacks instead.

Reply:
Done 🫡